File: 1865-finding-pairs-with-a-certain-sum/1865-finding-pairs-with-a-certain-sum.cpp
class FindSumPairs {
private:
    vector<int> nums1, nums2;
    unordered_map<int, int> freq2;

public:
    FindSumPairs(vector<int>& nums1, vector<int>& nums2) {
        this -> nums1 = nums1;
        this -> nums2 = nums2;
        for (int num : nums2) ++freq2[num];
    }
    
    void add(int index, int val) {
        --freq2[nums2[index]];
        nums2[index] += val;
        ++freq2[nums2[index]];
    }
    
    int count(int tot) {
        int pairs = 0;
        for (int i = 0; i < nums1.size(); ++i){
            int target = tot - nums1[i];
            pairs += freq2[target];
        }

        return pairs;
    }
};

/**
 * Your FindSumPairs object will be instantiated and called as such:
 * FindSumPairs* obj = new FindSumPairs(nums1, nums2);
 * obj->add(index,val);
 * int param_2 = obj->count(tot);
 */

File: 1865-finding-pairs-with-a-certain-sum/1865-finding-pairs-with-a-certain-sum.py
class FindSumPairs:

    def __init__(self, nums1: List[int], nums2: List[int]):
        self.nums1, self.nums2 = nums1, nums2
        self.freq2 = Counter(nums2)
        self.n1 = len(nums1)

    def add(self, index: int, val: int) -> None:
        self.freq2[self.nums2[index]] -= 1
        self.nums2[index] += val
        self.freq2[self.nums2[index]] += 1

    def count(self, tot: int) -> int:
        pairs = 0
        for i in range(self.n1):
            target = tot - self.nums1[i]
            pairs += self.freq2[target]

        return pairs


# Your FindSumPairs object will be instantiated and called as such:
# obj = FindSumPairs(nums1, nums2)
# obj.add(index,val)
# param_2 = obj.count(tot)